com.unity.ml-agents/Runtime/DemonstrationRecorder.cs

```diff
@@ -12,23 +12,24 @@ namespace MLAgents
     [AddComponentMenu("ML Agents/Demonstration Recorder", (int)MenuGroup.Default)]
     public class DemonstrationRecorder : MonoBehaviour
     {
+        [Tooltip("Whether or not to record demonstrations.")]
         public bool record;
+
+        [Tooltip("Base demonstration file name. Will have numbers appended to make unique.")]
         public string demonstrationName;
-        string m_FilePath;
+
+        [Tooltip("Base directory to write the demo files. If null, will use {Application.dataPath}/Demonstrations.")]
+        public string demonstrationDirectory;
+
         DemonstrationStore m_DemoStore;
-        public const int MaxNameLength = 16;
+        internal const int MaxNameLength = 16;
 
-        void Start()
-        {
-            if (Application.isEditor && record)
-            {
-                InitializeDemoStore();
-            }
-        }
+        const string k_ExtensionType = ".demo";
+        IFileSystem m_FileSystem;
 
         void Update()
         {
-            if (Application.isEditor && record && m_DemoStore == null)
+            if (record)
             {
                 InitializeDemoStore();
```

Comment:
initialization is called every update. I would recommend calling it LazyInitialization.